tick/optim/model/src/poisreg.cpp

```diff
@@ -71,12 +76,101 @@ double ModelPoisReg::sdca_dual_min_i(const ulong i,
   // Check we are in the correct bounds
   if (new_dual >= label) {
     new_dual = label - epsilon;
-    delta_dual = new_dual - dual;
+    delta_dual = new_dual - dual_i;
   }
 
   return delta_dual;
 }
 
+void ModelPoisReg::init_non_zero_label_map() {
+  non_zero_labels = VArrayULong::new_ptr();
+  for (ulong i = 0; i < get_n_samples(); ++i) {
+    if (get_label(i) != 0) {
+      non_zero_labels->append1(i);
+    }
+  }
+  n_non_zeros_labels = non_zero_labels->size();
+  non_zero_label_map_computed = true;
+}
+
+double ModelPoisReg::sdca_dual_min_i_identity(const ulong i,
+                                              const double dual_i,
+                                              const ArrayDouble &primal_vector,
+                                              const double previous_delta_dual_i,
+                                              double l_l2sq) {
+
+  if (fit_intercept) {
+    TICK_ERROR("SDCA not implemented with fit intercept yet");
```

Comment:
Yes, I think we should write tests / examples and verify that it's working. Can you post your code in which the intercept was not properly retrieved?